Rick and Morty Project

kabir-webDev opened the pull request:
Done List:

Homepage:
1. list 20 characters per page
2. have pagination
3. each character should have a card

Card:
1. image
2. name
3. a link to the character's profile

Profile:
1. species
2. type
3. gender
4. location name
5. backlink to the homepage

Search:
1. A search box should be present on all pages.
2. After the user submits the search query, redirect the user to this route: /search?q=%s
3. Then show a listing of found results. 

Nice-to-Haves:
1. Use SEO-friendly page slugs. E.g. /profile/24-rick-sanchez
2. Set page title for each different page
3. Submit the search form when the user presses the Enter key

choyan commented:
1. There are ESLint errors in several places.
2. If I reload the profile page, the page stucked at the loading state. 
3. It looks like you are filtering the data from the first page only. Not cultivating the backend filters. "https://rickandmortyapi.com/documentation/#filter-characters" 
4. Recently Visited Profiles are not available yet. 
5. We use 'React-query' with our projects. Can you use it and integrate it here?

kabir-webDev commented:
1. Yes, I actually didn't set ESLint up. That's why the errors are there.
2. I was routing to the profile page based on the redux data. That's why when the page is reloading, all data is gone and the page stucked at the loading state. But I can fetch the data based on ID directly from the API if needed. I just thought the process I implemented, might work. 
3. Yes, I was filtering the data from the current page. Actually, it was my bad that, I didn't ask about it. I thought all I have is an API from which I'm getting 20 data from each page. I should've been more careful.
4. I tried to do this using the localstorage. Added timestamp to keep track and filter last 10 visited data. But something went wrong and I couldn't fix that. I've spent most of my time, trying to solve this issue but couldn't.
5. I did use React-query a few times before. I think I can use it and integrate it here.
